modules/codegen/src/main/scala/com/twilio/guardrail/generators/CirceProtocolGenerator.scala

```diff
@@ -90,7 +90,13 @@ object CirceProtocolGenerator {
   object ModelProtocolTermInterp extends (ModelProtocolTerm ~> Target) {
     def apply[T](term: ModelProtocolTerm[T]): Target[T] = term match {
       case ExtractProperties(swagger) =>
-        Target.pure(Either.fromOption(Option(swagger.getProperties()).map(_.asScala.toList), "Model has no properties"))
+        Target.pure(
+          (swagger match {
+            case m: ModelImpl        => Option(m.getProperties)
+            case comp: ComposedModel => comp.getAllOf().asScala.toList.get(1).flatMap(prop => Option(prop.getProperties))
```

Comment:
What's the `get(1)` referring to? By writing things this way, we're only permitting `allOf` with two members, where the first member is the superclass, and the second is a model whose properties get appended to the first member?

If this is the case, is it possible to `.traverse(prop => Option(prop.getProperties))` to accumulate all union members? (I didn't see any tests for combining more than two models together; If I'm misreading and this is actually already possible, my apologies)

Reply:
Sorry. I've wrote it before. `allOf` contains as first element own model and as second one superclass. In the swagger documentation no word about multiple inheritance.

Reply:
Ah, yes; we talked about it in one of the other PRs. Thank you for explaining again.

Still, I'm wondering if something like:
```yaml
	  Cat:
    description: A representation of a cat
    allOf:
    - $ref: '#/definitions/Pet'
    - $ref: '#/definitions/Mammal'
    - type: object
      properties:
        huntingSkill:
          type: string
          description: The measured skill for hunting
          default: lazy
          enum:
          - clueless
          - lazy
          - adventurous
          - aggressive
```
would continue to add to this list. Could a dynamic approach be taken instead of hard-coding `1` here?

Reply:
I can't give you quickly answer. I need time to research.